src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +106,62 @@ public Object instantiate() throws ConstructionException {
         } catch (IllegalAccessException e) {
             throw new ConstructionException(ctor, "Access violation on " + ctor, e);
         }
+        finally {
+            mdcContext.finish();
+        }
 
         // satisfy dependencies in the order of the list, which was
         // prepared to comply with JSR 330
         Map<Method, InjectionArgs> settersAndArguments = new HashMap<Method, InjectionArgs>();
         for (Desire d: desires) {
+            Object value;
+            FieldInjectionPoint fd = null;
+            Field field;
+            LogContext mdcContextInjectionPointFieldValue = null;
             if (d.getInjectionPoint() instanceof FieldInjectionPoint) {
-                FieldInjectionPoint fd = (FieldInjectionPoint) d.getInjectionPoint();
-                Object value = checkNull(fd, providers.get(d).instantiate());
-                Field field = fd.getMember();
-
                 try {
+                    mdcContextInjectionPointFieldValue = LogContext.create();
```

Comment:
We should declare & initialize this variable right before `try`. Also, the variable name is a little long.